doc/book/sql-ddl.md

```diff
@@ -87,8 +87,21 @@ $table->changeColumn('name', Column\Varchar('new_name', 50));
 You may also *drop* existing columns or constraints:
 
 ```php
+use \Zend\Db\Metadata\Object\ConstraintObject;
+
 $table->dropColumn('foo');
-$table->dropConstraint('my_index');
+
+$constraint = new ConstraintObject('my_index', null);
+$table->dropConstraint($constraint);
+```
+
+Notice: On MySQL, you need to specify the type of constraint you want to drop:
+```php
+use \Zend\Db\Metadata\Object\ConstraintObject;
+
+$fkConstraint = new ConstraintObject('my_fk', null);
+$fkConstraint->setType('FOREIGN KEY');
```

Comment:
Was about to suggest to let  `dropConstraint` method take in `AbstractConstraint` so you have access to best of both `ConstraintObject` and `ForeignKey`, but what do you know, `ConstraintObject` does not inherit from `AbstractConstraint` as name would imply!
Just cannot win here without
>rewrite half of the framework either.
But that's too much to ask for at this time ...

Great job on moving specifications to decorators. Those are tough to do.

For version 3, which according to discussions all new versions want to target PHP 7.1 strict type system, it would be needed to rewrite to tighten up the types allowed per method to weed out those 
>nasty inconsistencies to take care of ahead of that bug.
I would have fun doing it, but a little discouraging based on rate of PR reviews.